src/averageAges.js

```diff
@@ -34,7 +44,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = (withChildren !== undefined)
+    ? people.filter(person => {
+      return person.sex === 'f'
+             && people.some(x => x.mother === person.name);
+    })
+    : people.filter(person => person.sex === 'f');
```

Comment:
```suggestion
  const women = people.filter(person =>
    withChildren
      ? people.some(child => child.mother === person.name) && person.sex === 'f'
      : person.sex === 'f'
  );
```

Same story as with 'men'